src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    const type = action.type; // string with type property
+    const addValues = action.extraData; // object with extra data
+    const removeValues = action.keysToRemove; // array with remove keys
```

Comment:
use object destructuring
```suggestion
    const { type, extraData, keysToRemove } = action;
```